### ISSUE TITLE
CA2007: output_kind not inheritable through directories

### ISSUE THREAD

**Issue by wilhelmzapiain** [dotnet/roslyn-analyzers#3919]
#### Analyzer package

[Microsoft.CodeAnalysis.FxCopAnalyzers](https://www.nuget.org/packages/Microsoft.CodeAnalysis.FxCopAnalyzers)

#### Package Version

v3.0.0 (Latest)
Visual Studio Community 16.6.5

#### Diagnostic ID

[CA2007](https://docs.microsoft.com/en-us/visualstudio/code-quality/ca2007)

#### Repro steps

I have an example made in https://github.com/wilhelmzapiain/CA2007BugDemo

1. Create a new Console App (.Net Core). Make sure the "_Place solution and project in the same directory_" box is **unchecked**.
2. Selecting the Solution, Add a New Item, editorconfig File (default).
3. Add the next line to the .editorconfig file:
   `dotnet_code_quality.CA2007.output_kind = DynamicallyLinkedLibrary`
4. Add a reference to the Microsoft.CodeAnalysis.FxCopAnalyzers package.
5. The Program.cs file should contain the following:
    ```C#
    using System;
    using System.Threading.Tasks;
    
    namespace CA2007BugDemo
    {
        public static class Program
        {
            public static async Task Main()
            {
                await Task.Delay(500);
                Console.WriteLine("Hello World!");
            }
        }
    }
    ```

#### Expected behavior

No warning in the `await Task.Delay(500);` line. The .editorconfig in the poarent folder says to check only .dll's for CA2007.

#### Actual behavior

Visual Studio warns of CA2007, suggesting to Add .ConfigureAwait to the Delay, bypassing the instruction in the .editorconfig file to only check .dll's.
Also, if the project file is in the same directory as the .editorconfig file, the warning does not appear, so it appears as if this setting is not inheritable. In my demo project CA2007BugDemo is in the same directory and does not show the warning, while project CA2007BugDemo2 is in another directory and shows the warning.

**Comment by mavasani** [comment]
Can you please try latest 3.3.x-beta packages? They support the requested configuration. See the 2 sections at https://docs.microsoft.com/en-us/visualstudio/code-quality/configure-fxcop-analyzers?view=vs-2019#enabling-editorconfig-based-configuration

**Comment by wilhelmzapiain** [comment]
Now both projects ignore the setting and show a warning when they shouldn't.

**Comment by Evangelink** [comment]
@wilhelmzapiain I have managed to reproduce and solve your issue. In your `.editorconfig` file you need to add `[*.cs]` before the analyzer configuration and make sure to be on at least v3.3.0
